packages/tldraw/src/lib/shapes/text/toolStates/Pointing.ts

```diff
@@ -168,12 +168,22 @@ export class Pointing extends StateNode {
 			const transform = this.editor.getShapeParentTransform(shape)
 			delta.rot(-transform.rotation())
 		}
-
-		this.editor.updateShape({
-			...shape,
-			x: shape.x + delta.x,
-			y: shape.y + delta.y,
-		})
+		const isGridMode = this.editor.getInstanceState().isGridMode
```

Comment:
btw you could consider moving all this logic above line 144 - because most of this logic is about calculating `delta`

Reply:
Will do!

Reply:
I actually want to calculate the delta as normal and then shift the top left of the shape to the nearest grid point, this keeps the text tool working in a similar way in grid mode and outside of it.